CMakeLists.txt

```diff
@@ -94,7 +83,44 @@ else()
   set(CMAKE_EXE_LINKER_FLAGS_INIT "${CMAKE_EXE_LINKER_FLAGS_INIT} -Wl,-z,defs")
 endif()
 
-add_subdirectory(src)
+set(IRODS_ENABLE_GENQUERY2 OFF CACHE BOOL "Enable GenQuery2 support. GenQuery2 package must be installed.")
+
+set(CMAKE_IRODS_HTTP_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
My question about the top-level object target names raised another question.

Does `CMAKE_` here carry any special rules? Why would we use `CMAKE_` as the prefix rather than `IRODS_`?

Reply:
We use similar variables in the irods project that start with `CMAKE_IRODS_`. I think I did this to match `CMAKE_CURRENT_SOURCE_DIR` and `CMAKE_CURRENT_BINARY_DIR`. I'm not married to it though.

Reply:
All good. I was just wondering. I don't have any strong opinions on it at this time.

Reply:
On further reflection, and after a quick peek at the [CMake docs](https://cmake.org/cmake/help/v3.28/manual/cmake-language.7.html#variables), I think I am going to change this. I don't think we should be declaring our own `CMAKE_` variables, at least not for this case.

Reply:
I agree.